src/Components/pagination.tsx

```diff
@@ -0,0 +1,81 @@
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+
+type Props = {
+  currentPage: number
+  totalCount: number
+  perPage: number
+}
+
+function getPageNumbers(currentPage: number, totalPage: number): Array<string | number> {
+  // 6ページ以下の場合は全て表示する
+  if (totalPage <= 6) {
+    return Array.from({ length: totalPage }, (_, index) => index + 1)
+  }
+
+  if (currentPage <= 3) {
+    return [1, 2, 3, 4, 5, '...']
+  }
+
+  if (currentPage >= totalPage - 2) {
+    return ['...', totalPage - 4, totalPage - 3, totalPage - 2, totalPage - 1, totalPage]
+  }
+
+  return [
+    '...',
+    currentPage - 2,
+    currentPage - 1,
+    currentPage,
+    currentPage + 1,
+    currentPage + 2,
+    '...',
+  ]
+}
+
+export default function Pagination({ currentPage, totalCount, perPage }: Props) {
+  const router = useRouter()
+  let totalPage = Math.ceil(totalCount / perPage)
+  // 1000件以上はAPI制限でエラーになるため、1000件までの遷移先しか表示しない
+  if (totalCount > 1000) {
```

Comment:
説明変数を作ってください。
コメントが添えられているだけマシですが、マジックナンバーは徹底的に排除してください。
(1行だけの短命であればよいですが下の40行目でも再利用しているということは、いずれどこかで再利用される可能性が高いため)

```suggestion
// 1000件以上はAPI制限でエラーになるため、1000件までの遷移先しか表示しない
const apiMaxCount = 1000
  if (totalCount > apiMaxCount) {
```

あと細かいですが、コメントだと1000件目もエラーになりそうなものですが、
・totalCount >= 1000かつ1000件以上はAPI制限でエラーになるため、999件までの遷移先しか表示しない
・totalCount >1000かつ1001件以上はAPI制限でエラーになるため、1000件までの遷移先しか表示しない
のどちらが正しいですか？
(のように誤解を生むもとになるので、私は説明変数が好きです。コードが全てなので。)

Reply:
誤解を生む書き方をしてしまい申し訳ありません🙇
正しくは
- totalCount >1000かつ1001件以上はAPI制限でエラーになるため、1000件までの遷移先しか表示しない

です